StreamChatClient_v3/Controllers/Controller.swift

```diff
@@ -27,20 +53,16 @@ public extension ControllerRemoteActivityDelegate {
 }
 
 /// A helper protocol allowing calling delegate on a given dispatch queue.
-protocol DelegateCallbable {
+protocol DelegateCallable {
     associatedtype Delegate
     var anyDelegate: Delegate { get }
 }
 
-extension DelegateCallbable where Self: Controller {
+extension DelegateCallable where Self: Controller {
+    /// A helper function to ensure the delegate callback is performed on the callback queue.
     func delegateCallback(_ callback: @escaping (Delegate) -> Void) {
-        if callbackQueue == DispatchQueue.main, Thread.current.isMainThread {
-            // Call the delegate on the main queue synchronously
-            callback(anyDelegate)
-        } else {
-            callbackQueue.async {
-                callback(self.anyDelegate)
-            }
+        self.callback {
+            callback(self.anyDelegate)
```

Comment:
if this is just going to use `callback` from controller... why do we need `callback` and `delegateCallback`?

Reply:
Haha, I knew you were going to ask about this! :D I'm not sure about this change, to be honest. Reasons I kept it:
- I kinda like that when you use it, it's clear that it's related to the delegate callback. 
- You don't have to write `self.delegate`
- In the future, we don't have to call this at all, if no delegate is assigned
All these reasons are quite weak, but it was enough for me to have it here. Again, we can delete in the future if we see it brings no additional value over `callback { }`